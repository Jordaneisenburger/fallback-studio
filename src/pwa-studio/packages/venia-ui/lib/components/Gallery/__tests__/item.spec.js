import React from 'react';
import { shallow } from 'enzyme';
import { Link, MemoryRouter } from 'react-router-dom';

import Item from '../item';

const classes = {
    image: 'a',
    image_pending: 'b',
    imagePlaceholder: 'c',
    imagePlaceholder_pending: 'd',
    images: 'e',
    images_pending: 'f',
    name: 'g',
    name_pending: 'h',
    price: 'i',
    price_pending: 'j',
    root: 'k',
    root_pending: 'l'
};

const validItem = {
    id: 1,
    name: 'Test Product',
    small_image: '/foo/bar/pic.png',
    url_key: 'strive-shoulder-pack',
    price: {
        regularPrice: {
            amount: {
                value: 21,
                currency: 'USD'
            }
        }
    }
};

/**
 * STATE 0: awaiting item data
 * `item` is `null` or `undefined`
 */
test('renders a placeholder item while awaiting item', () => {
    const wrapper = shallow(<Item classes={classes} />).dive();
    const child = wrapper.find('ItemPlaceholder');
    const props = { classes };

    expect(child).toHaveLength(1);
    expect(child.props()).toMatchObject(props);
});

test('passes classnames to the placeholder item', () => {
    const wrapper = shallow(<Item classes={classes} />)
        .dive()
        .dive();

    expect(wrapper.hasClass(classes.root_pending));
});

test('renders Link elements for navigating to a PDP', () => {
    const wrapper = shallow(
        <MemoryRouter>
            <Item classes={classes} item={validItem} />
        </MemoryRouter>
    );

    expect(
        wrapper
            .childAt(0)
            .dive()
            .dive()
            .findWhere(
                node =>
                    node.type() === Link &&
                    node.prop('to') === `/${validItem.url_key}.html`
            )
    ).toHaveLength(2);
});

/**
 * STATE 1: ready
 * `item` is a valid data object
 */
test('renders only the real image', () => {
    const wrapper = shallow(<Item classes={classes} item={validItem} />).dive();
    const image = wrapper.find({ className: classes.image });

    expect(image).toHaveLength(1);
});
