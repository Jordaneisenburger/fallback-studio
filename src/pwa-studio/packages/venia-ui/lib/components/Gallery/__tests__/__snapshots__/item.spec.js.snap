// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a placeholder item while awaiting item 1`] = `
<div
  className="root_pending"
>
  <div
    className="images_pending"
  >
    <div
      className="root container"
    >
      <img
        alt="Placeholder for gallery item image"
        className="image placeholder_layoutOnly"
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        width={100}
      />
      <img
        alt="Placeholder for gallery item image"
        className="image loaded"
        loading="lazy"
        onError={[MockFunction]}
        onLoad={[MockFunction]}
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
      />
    </div>
  </div>
  <div
    className="name_pending"
  />
  <div
    className="price_pending"
  />
</div>
`;

exports[`renders correctly with valid item data 1`] = `
<div
  className="root"
>
  <a
    className="images"
    href="/strive-shoulder-pack.html"
    onClick={[Function]}
  >
    <div
      className="root container"
    >
      <img
        alt="Test Product"
        className="image placeholder_layoutOnly"
        height={375}
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        width={100}
      />
      <img
        alt="Test Product"
        className="image loaded"
        height={375}
        loading="lazy"
        onError={[MockFunction]}
        onLoad={[MockFunction]}
        sizes="(max-width: 640px) 300px, 840px"
        src="/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=100&height=375&fit=cover"
        srcSet="/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
/media/catalog/product/foo/bar/pic.png?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
        width={100}
      />
    </div>
  </a>
  <a
    className="name"
    href="/strive-shoulder-pack.html"
    onClick={[Function]}
  >
    <span>
      Test Product
    </span>
  </a>
  <div
    className="price"
  >
    <div />
  </div>
</div>
`;
